sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+import copy
+
+import astropy.units as u
+import numpy as np
+from scipy import interpolate, stats
+from sunpy.data import manager
+
+from sunxspex.io import load_chianti_continuum, load_chianti_lines_lite, load_xray_abundances
+
+doc_string_params = """
+Parameters
+----------
+energy_edges: `astropy.units.Quantity`
+    The edges of the energy bins in a 1D N+1 quantity.
+
+temperature: `astropy.units.Quantity`
+    The temperature of the plasma.
+    Can be scalar or 1D of any length. If not scalar, the flux for each temperature
+    will be calculated. The first dimension of the output flux will correspond
+    to temperature.
+
+emission_measure: `astropy.units.Quantity`
+    The emission measure of the plasma at each temperature.
+    Must be same length as temperature or scalar.
+
+abundance_type: `str` (optional)
+    Abundance type to use.  Options are:
+        1. cosmic
+        2. sun_coronal - default abundance
+        3. sun_coronal_ext
+        4. sun_hybrid
+        5. sun_hybrid_ext
+        6. sun_photospheric
+        7. mewe_cosmic
+        8. mewe_solar
+    The values for each abundance type is stored in the global
+    variable DEFAULT_ABUNDANCES which is generated by `setup_default_abundances`
+    function. To load different default values for each abundance type,
+    see the docstring of that function.
+
+relative_abundances: `tuple` of `tuples` of (`int`, `float`) (optional)
+    The relative abundances of different elements as a fraction of their
+    default abundances defined by abundance_type.
+    Each tuple represents the values for a given element.
+    The first entry represents the atomic number of the element.  If this is not
+    entered as an int, it is rounded to the nearest int.
+    The second entry represents the axis represents the fraction by which the
+    element's default abundance should be scaled.
+
+observer_distance: `astropy.units.Quantity` (Optional)
+    The distance between the source and the observer. Scales output to observer distance
+    and unit by 1/length.
```

Comment:
I'm not sure what this means like the output is scaled to the input units e.g. km or au? In my head that would be a 1/length square term ... but I'm probably wrong.

Reply:
No, you're absolutely right.